### ISSUE TITLE
Disable apostrophe buttons with the `disabled` attribute

### ISSUE THREAD

**Issue by abea** [apostrophecms/apostrophe#1472]
I added a `danger` button to my editor module, but wanted it disabled except under certain conditions. I added the `disabled` attribute easily enough, but there was no visual indication of that. I then found the `--disabled` class for styling, so I used the base macro instead of the danger macro so I could add the second class.

Now in client JS I have to remove both the class and the attribute to enable it. **My suggestion is that since buttons are disabled with an attribute in the HTML spec, we apply disabling styles with only that attribute rather than require the second class.**

There's [a good post from the EBay tech blog](https://www.ebayinc.com/stories/blogs/tech/how-our-css-framework-helps-enforce-accessibility/) about their approach to this. They point to how it helps with enforcing good, accessible HTML practices.

I think it'd be worth leaving the disabled button _class_ out in 3.x until really necessary. The only need should be in an `a` tag is absolutely needed to do the job, but for cases that involve disabling may all be satisfied with `button`s.

**Comment by houmark** [comment]
I noticed that also and not being disabled allows the user to click. I have a PR in the making that improves that without using disabled, which also came to my mind in the process of considering solutions for how buttons etc. are working now.
